internal/services/web/static_site_resource.go

```diff
@@ -216,6 +234,20 @@ func resourceStaticSiteRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	}
 	d.Set("api_key", apiKey)
 
+	appSettingsResp, err := client.ListStaticSiteAppSettings(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(appSettingsResp.Response) {
+			log.Printf("[DEBUG] Application Settings of %s were not found", id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on %s: %+v", id, err)
+	}
```

Comment:
should we just return an error here rather than removing from the state?

Reply:
Good shout!